evaluate/task/migrate.go

```diff
@@ -0,0 +1,199 @@
+func clearRepositoryForMigration(repositoryPath string, language language.Language, allFilePaths []string, testFilePath string) (err error) {
+	testFileName := filepath.Base(testFilePath)
+	testFileName = strings.TrimSuffix(testFileName, language.DefaultTestFileSuffix())
+
+	for _, filePath := range allFilePaths {
+		if filePath == testFilePath {
+			continue
+		}
+
+		fileName := filepath.Base(filePath)
+		fileName = strings.TrimSuffix(fileName, language.DefaultFileExtension())
+		if fileName == testFileName {
+			continue
+		}
+
+		if err := os.Remove(filepath.Join(repositoryPath, filePath)); err != nil {
+			return pkgerrors.WithStack(err)
+		}
+	}
+
+	return nil
+}
+
+// validateTranspileRepository checks if the repository for the "transpile" task is well-formed.
+func validateMigrateRepository(logger *log.Logger, repositoryPath string, language language.Language) (err error) {
+	logger.Printf("validating repository %q", repositoryPath)
+
+	filePaths, err := osutil.FilesRecursive(repositoryPath)
+	if err != nil {
+		return pkgerrors.WithStack(err)
+	}
+
+	// Keep a mapping between implementation file paths and test file paths.
+	implementationFileNames := map[string]bool{}
+	testFileNames := map[string]bool{}
+	for _, filePath := range filePaths {
+		filePathExtension := filepath.Ext(filePath)
+		// Ignore build and configuration files.
+		if filePathExtension == ".xml" || filePathExtension == ".json" {
+			continue
+		} else if filePathExtension != language.DefaultFileExtension() {
+			return pkgerrors.Errorf("the repository %q must contain only %s files but found %q", repositoryPath, language.Name(), filePath)
+		}
+
+		fileName := filepath.Base(filePath)
+		if strings.HasSuffix(filePath, language.DefaultTestFileSuffix()) {
+			fileName = strings.TrimSuffix(fileName, language.DefaultTestFileSuffix())
+			testFileNames[fileName] = true
+		} else {
+			fileName = strings.TrimSuffix(fileName, language.DefaultFileExtension())
+			implementationFileNames[fileName] = true
+		}
+	}
+
+	if len(implementationFileNames) == 0 {
+		return pkgerrors.Errorf("the repository %q must contain implementation files but found none", repositoryPath)
+	} else if len(testFileNames) == 0 {
+		return pkgerrors.Errorf("the repository %q must contain test files but found none", repositoryPath)
+	}
+
+	// Check if for each implementation file a test file exists.
+	for implementationFileName := range implementationFileNames {
+		if !testFileNames[implementationFileName] {
+			return pkgerrors.Errorf("the repository %q must contain a test file for each implementation file but found none for %q", repositoryPath, implementationFileName)
+		}
+	}
+
+	// Check if for each test file an implementation file exists.
+	for testFileName := range testFileNames {
+		if !implementationFileNames[testFileName] {
+			return pkgerrors.Errorf("the repository %q must contain an implementation file for each test file but found none for %q", repositoryPath, testFileName)
+		}
+	}
```

Comment:
Would maybe track them in two slices and then sort them and use [`slices.Equal`](https://pkg.go.dev/slices#Equal).

Reply:
I would still change cause it leaves less room for errors and is easier to read. Let's go clean code > usability here.